src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+  selectedCardId = cardId;
+  openModal(cardDeleteModal);
+}
+
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button-liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button-liked");
+    })
+    .catch(console.error);
+}
+
+//using data as a placeholder but they are objects, precisely the ones above to clone cards
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button-liked");
+  }
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.setAttribute("src", data.link);
+  cardImageEl.setAttribute("alt", data.name);
+  // Can also be written as:
+  // cardImageEl.src = data.link
+  // cardImageEl.alt = data.alt
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+previewCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+function handleEscKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    closeModal(openedModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const cardSubmitBtn = evt.submitter;
+  setButtonText(cardSubmitBtn, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save");
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+  setButtonText(avatarSubmitBtn, true, "Saving...");
+
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      avatar.src = data.avatar;
+      closeModal(avatarModal);
+      evt.target.reset();
```

Comment:
* You should **disable** the `avatar` submit button after submission as well https://snipboard.io/ilSakD.jpg